### ISSUE TITLE
[Question] Translate autosklearn hyperparameter to sklearn ensemble classifier 

### ISSUE THREAD

**Issue by JohnWestlund** [automl/auto-sklearn#1637]
# Short Question Description
I'm trying to take the hyperparameters identified by autosklearn and setting them for a sklearn classifier. In this case `gradient_boosting` to `sklearn.ensemble.GradientBoostingClassifier`

For example, I see these hyperparameters listed:
![image](https://user-images.githubusercontent.com/5503935/210494702-2ceb1198-3b55-414d-b596-1186723cb214.png)

I can see how something like `early_stop=valid` might translate to `tol` and `n_iter_no_change` -- but a `tol` of 0.0000 makes it look like it's not enabled.

Is there a mapping?

**Comment by aron-bram** [comment]
Hi,

your assumption about "early_stopping" is correct. When it is set to "valid", the estimator behaves similarly to the sklearn implementation of early stopping.

However, the tolerance parameter that you see there is likely set to 1e-7, as this is a hyperparameter that we keep constant during the optimization. In sklearn this is set to 1e-4 by default.
Could it be that the value column in your screenshot trims the content of each cell to 6 characters?
This would explain why it shows "0.00000" instead of "0.0000001" (1e-7). 

Sorry, I'm not quite sure if I understood your last question. What kind of mapping would you like to have other than the one you showcased?

**Comment by JohnWestlund** [comment]
Thanks Aron. That was my assumption as well regarding the display of "0.00000". Looking at the Hyperparameters above: 
`early_stop`, `l2_regularization`, `learning_rate`, `loss`, `max_bins`, `max_depth`, `max_leaf_nodes`, `min_sample_leaf`, `n_iter_no_change`, `scoring`, `tol` and `validation_fraction` 

and comparing them with the [SKLearn Gradient Boosting](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.GradientBoostingClassifier.html)  parameters, there's not an equivalent hyperparameter for `early_stop` (but like we mentioned there is a similar behavior from other hyperparameters), `loss` (there is no `auto` value), `max_bin`, and `scoring`.

So I can instantiate a `GradientBoostingClassifier` with some of the hyperparamters:

```
h_gb = GradientBoostingClassifier(n_estimators=300,
                                learning_rate=0.01083,
                                max_depth=None,
                                max_leaf_nodes=25,
                                min_samples_leaf=4,
                                n_iter_no_change=19,
                                validation_fraction=0.1759)
```
but it's not clear what to do with the others.

Hopefully that clarifies my question. Thanks for taking the time to respond.

**Comment by aron-bram** [comment]
Thank you for the clarification.

We are currently using a [HistGradientBoostingClassifier](https://scikit-learn.org/0.24/modules/generated/sklearn.ensemble.HistGradientBoostingClassifier.html#sklearn.ensemble.HistGradientBoostingClassifier) from sklearn instead of a [GradientBoostingClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.GradientBoostingClassifier.html#sklearn.ensemble.GradientBoostingClassifier), which you are referring to.
(However, your assumption of "gradient_boosting" corresponding to the regular gradient boosting classifier is more rational, therefore apologies for the inappropriate naming)

You might notice that the first link refers to an older implementation of the classifier, namely version 0.24 of sklearn. That is because as of now, we depend on that version. We are currently working on updating our dependency to the latest stable version of sklearn, namely 1.2. There are also simultaneous dependency updates that break our code, so the realease will take some time.

So to briefly answer you question, you can directly translate these hyperparameters that you mentioned above to a HistGradientBoostingClassifier of version 0.24 in sklearn. I believe, the parameters that you can't see there, are left as default. Apart from the "warm_start" parameter, which is set to True.

I hope I could help you. Please let me know if something is still unclear.

Best,
Aron

**Comment by JohnWestlund** [comment]
Thanks, Aron. That answers my question. Thanks for clearing up the confusion.
